


<!-- PROJECT LOGO -->
<br />
<p align="center">
  <a href=>
  <h3 align="center"> Balanced SET levels favor the correct enhancer repertoire during cell fate acquisition
</h3>
  </a>
  <p align="center">
    Mattia Zaghi, Federica Banfi, Luca Massimino, Monica Volpin, Edoardo Bellini, Simone Brusco, Ivan Merelli, Cristiana Barone, Cristina Sironi, Michela Bruni, Giulia Fagnocchi, Alicia Rubio-Garrido, Linda Bossini, Luigi Lamparelli, Laura Pintado, Deborah D'Aliberti, Rocco Piazza, Luca Mologni, Gaia Colasante, Federica Ungaro, Francesco Ferrari, Jean-Michel Cioni, Emanuele Azzoni, Rocco Piazza, Eugenio Montini, Vania Broccoli and Alessandro Sessa
  </p>
</p>



<!-- TABLE OF CONTENTS -->
<details open="open">
  <summary>Table of Contents</summary>
  <ol>
  <li>
      <a href="#About the paper">About the paper</a>
  </li>
    <li>
      <a href="#pipelines">Pipelines</a>
    </li>
    <li>
      <a href="#figures">Figures</a>
    </li>
  </ol>
</details>



<!-- About the paper -->
## About the paper

In this work we described the effect of SET protein accumulation on chromatin rewiring in vitro, using Schinzel-Giedion syndrome patients IPSCs, and in vivo using Mouse and Zebrafish model. We used a multiomic approach combining ATAC-seq, ChIP-seq, Hi-C, RNA-seq and scMultiome (ATAC+RNA) to address the question on a genomic standpoint. In this page all the codes and pipelines used to analyze all data and produce the manuscript figures are deposited. (I.E the page is under construction as of now, all the scripts will be uploaded in the next days).

<!-- Pipelines -->
## Pipelines
In this folder the snakemake pipeline used to analyzed all the epigenomic data (ATAC, ChIP-seq, Hi-C) are deposited. Within every folder are present:
* `.sk` file, that contain all rules sequence.

* `.yaml` file, that must be edit according to organism and statistics and user needs.

<!-- Figures -->
## Figures
In scripts folder are present all the costum snakemake and R script used to generate plots and figures. Each figure folder contain the code used to generate it and its related supplementary figure.

<!-- CONTACT -->
## Contacts

Mattia Zaghi - <zaghi.mattia@hsr.it>

Edoardo Bellini - <bellini.edoardo@hsr.it>

Alessandro Sessa - <sessa.alessandro@hsr.it>

<!-- MARKDOWN LINKS & IMAGES -->
<!-- https://www.markdownguide.org/basic-syntax/#reference-style-links -->
[contributors-shield]: https://img.shields.io/github/contributors/othneildrew/Best-README-Template.svg?style=for-the-badge
[contributors-url]: https://github.com/othneildrew/Best-README-Template/graphs/contributors
[forks-shield]: https://img.shields.io/github/forks/othneildrew/Best-README-Template.svg?style=for-the-badge
[forks-url]: https://github.com/othneildrew/Best-README-Template/network/members
[stars-shield]: https://img.shields.io/github/stars/othneildrew/Best-README-Template.svg?style=for-the-badge
[stars-url]: https://github.com/othneildrew/Best-README-Template/stargazers
[issues-shield]: https://img.shields.io/github/issues/othneildrew/Best-README-Template.svg?style=for-the-badge
[issues-url]: https://github.com/othneildrew/Best-README-Template/issues
[license-shield]: https://img.shields.io/github/license/othneildrew/Best-README-Template.svg?style=for-the-badge
[license-url]: https://github.com/othneildrew/Best-README-Template/blob/master/LICENSE.txt
[linkedin-shield]: https://img.shields.io/badge/-LinkedIn-black.svg?style=for-the-badge&logo=linkedin&colorB=555
[linkedin-url]: https://linkedin.com/in/othneildrew
[product-screenshot]: images/screenshot.png
